src/pages/workspaces/workspace/Notebooks.js

```diff
@@ -357,8 +357,8 @@ const Notebooks = _.flow(
             style: {
               ...Style.elements.card.container, height: 125, marginTop: 15
             },
-            disabled: appIsSettingUp(app),
-            tooltip: appIsSettingUp(app) && 'Galaxy app is being created',
+            disabled: appIsSettingUp(app) || _.last(currentPersistentDiskIncludingUnattached(apps, galaxyDataDisks)),
+            tooltip: (appIsSettingUp(app) && 'Galaxy app is being created') || (_.last(currentPersistentDiskIncludingUnattached(apps, galaxyDataDisks)) && 'Your persistent disk is still attached to your previous app; you can create a new app once your previous app finishes deleting, which will take a few minutes.'),
```

Comment:
I wonder if it'd make the code easier to read if ```_.last(currentPersistentDiskIncludingUnattached(apps, galaxyDataDisks))``` were factored out so it reads something like

```javascript
disabled: appIsSettingUp(app) || isCurrentDiskDetaching(apps, galaxyDataDisks)
```

Reply:
I can make a helper method that calls `currentPersistentDiskIncludingUnattached` and returns the second value in the tuple. i don't want to change the name of this method since i use it in a couple of different places within this component

Reply:
AFAICS, all use cases depend on either `currentDataDisk` or `isCurrentDiskDetaching` so how about breaking up `currentPersistentDiskIncludingUnattached` a bit such that it returns only `currentDataDisk` as the name implies, and the remaining detaching status boolean is returned by `isCurrentDiskDetaching`, like so:

```javascript
export const currentPersistentDiskIncludingUnattached = (apps, galaxyDataDisks) => {
  // a user's PD can either be attached to their current app, detaching from a deleting app or unattached
  const currentGalaxyApp = currentAppIncludingDeleting(apps)
  const currentDataDiskName = currentGalaxyApp?.diskName
  const attachedDataDiskNames = _.without([undefined], _.map(app => app.diskName, apps))
  // if the disk is attached to an app (or being detached from a deleting app), return that disk. otherwise,
  // return the newest galaxy disk that the user has unattached to an app
  const currentDataDisk = currentDataDiskName ?
    _.find({ name: currentDataDiskName }, galaxyDataDisks) :
    _.last(_.sortBy('auditInfo.createdDate', _.filter(({ name, status }) => status !== 'Deleting' && !_.includes(name, attachedDataDiskNames), galaxyDataDisks)))
  return currentDataDisk
}
```

```javascript
const isCurrentGalaxyDiskDetaching = apps => {
  const currentGalaxyApp = currentAppIncludingDeleting(apps)    
  return currentGalaxyApp && (currentGalaxyApp.status === 'DELETING' || currentGalaxyApp.status === 'PREDELETING')
}
```